examples/regression/1-quickstart/plot_toy_model.py

```diff
@@ -5,53 +5,95 @@
+from numpy.typing import NDArray
+from sklearn.neural_network import MLPRegressor
 from mapie.metrics.regression import regression_coverage_score
 from mapie.regression import SplitConformalRegressor
 from mapie.utils import train_conformalize_test_split
 
-RANDOM_STATE = 42
+np.random.seed(42)
+
+##############################################################################
+# Firstly, let us create our dataset:
+
+
+def f(x: NDArray) -> NDArray:
+    """Polynomial function used to generate one-dimensional data."""
+    return np.array(5 * x + 5 * x**4 - 9 * x**2)
+
 
-X, y = make_regression(n_samples=500, n_features=1, noise=20, random_state=RANDOM_STATE)
+rng = np.random.default_rng()
+sigma = 0.1
+n_samples = 10000
+X = np.linspace(0, 1, n_samples)
+y = f(X) + rng.normal(0, sigma, n_samples)
 
-(
-    X_train, X_conformalize, X_test, y_train, y_conformalize, y_test
-) = train_conformalize_test_split(
-    X, y,
-    train_size=0.6, conformalize_size=0.2, test_size=0.2,
-    random_state=RANDOM_STATE
+(X_train, X_conformalize, X_test,
+ y_train, y_conformalize, y_test) = train_conformalize_test_split(
+    X, y, train_size=0.8, conformalize_size=0.1, test_size=0.1
 )
 
-confidence_level = [0.95, 0.68]
+##############################################################################
+# We fit our training data with a MLPRegressor.
+# Then, we initialize a :class:`~mapie.regression.SplitConformalRegressor`
+# using our estimator, indicating that it has already been fitted with
+# `prefit=True`.
+# Lastly, we compute the prediction intervals with the desired confidence level using
+# the ``conformalize`` and ``predict_interval`` methods.
+
+regressor = MLPRegressor(activation="relu")
+regressor.fit(X_train.reshape(-1, 1), y_train)
+
+confidence_level = 0.95
 mapie_regressor = SplitConformalRegressor(
-    confidence_level=confidence_level, prefit=False
+    estimator=regressor, confidence_level=confidence_level, prefit=True
 )
-mapie_regressor.fit(X_train, y_train)
-mapie_regressor.conformalize(X_conformalize, y_conformalize)
-y_pred, y_pred_interval = mapie_regressor.predict_interval(X_test)
+mapie_regressor.conformalize(X_conformalize.reshape(-1, 1), y_conformalize)
+y_pred, y_pred_interval = mapie_regressor.predict_interval(X_test.reshape(-1, 1))
+
+##############################################################################
+# ``y_pred`` represents the point predictions as a ``np.ndarray`` of shape
+# ``(n_samples)``.
+# ``y_pred_interval`` corresponds to the prediction intervals as a ``np.ndarray`` of
+# shape ``(n_samples, 2, 1)``, giving the lower and upper bounds of the intervals.
+
+##############################################################################
+# Finally, we can easily compute the coverage score (i.e., the proportion of times the
+# true labels fall within the predicted intervals).
 
-coverage_scores = regression_coverage_score(y_test, y_pred_interval)
+coverage_score = regression_coverage_score(y_test, y_pred_interval)
```

Comment:
Shall we print the coverage_score, and analyse quickly the result? Let's make sure it is not random (is MLPRegressor deterministic?)